src/hwmon.rs

```diff
@@ -0,0 +1,484 @@
+/* https://www.kernel.org/doc/Documentation/hwmon/sysfs-interface */
+/* The common scheme for files naming is: <type><number>_<item>. */
+
+use std::{collections::HashMap, path::PathBuf, str::FromStr};
+
+use itertools::Itertools;
+use thiserror::Error;
+use tokio::fs::DirEntry;
+
+use crate::{sysfs::npu_mgmt, DeviceError, DeviceResult};
+
+/// An error that occurred during parsing or retrieving hwmon sensors.
+#[derive(Debug, Error)]
+pub enum HwmonError {
+    #[error("Unsupported type: {name}")]
+    UnsupportedType { name: String },
+    #[error("No metric element file")]
+    NoMetricElementFile,
+    #[error("Not found item: {name}")]
+    NotFoundItemName { name: String },
+    #[error("Unexpected value format: {value}")]
+    UnexpectedValueFormat { value: String },
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+pub enum HwmonType {
+    Current,
+    Voltage,
+    Power,
+    Temperature,
+}
+
+impl FromStr for HwmonType {
+    type Err = HwmonError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "curr" => Ok(HwmonType::Current),
+            "in" => Ok(HwmonType::Voltage),
+            "power" => Ok(HwmonType::Power),
+            "temp" => Ok(HwmonType::Temperature),
+            _ => Err(HwmonError::UnsupportedType {
+                name: String::from(s),
+            }),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+struct HwmonMetricType {
+    hwmon_type: HwmonType,
+    idx: u8,
+}
+
+#[derive(Debug)]
+struct HwmonMetricItem {
+    item_name: String,
+    path: PathBuf,
+}
+
+#[derive(Debug)]
+struct HwmonMetricEntry {
+    metric_type: HwmonMetricType,
+    metric_item: HwmonMetricItem,
+}
+
+impl TryFrom<DirEntry> for HwmonMetricEntry {
+    type Error = HwmonError;
+
+    fn try_from(value: DirEntry) -> Result<Self, Self::Error> {
+        let filename = value.file_name().to_string_lossy().to_string();
+        let underscore_pos = filename.find('_').ok_or(HwmonError::NoMetricElementFile)?;
+
+        let (metric_type_str, metric_item_str) = filename.split_at(underscore_pos);
```

Comment:
```suggestion
        let (metric_type_str, metric_item_str) = filename.split_once('_').ok_or(HwmonError::NoMetricElementFile)?;
```
뭔가 비슷한 함수가 있을것 같아서 보았는데 split_once 를 활용할 수 있네요. 고려해봐 주시겠어요 ?

Reply:
딱 원하던 함수가 있었네요. 알려주셔서 감사합니다.